solidity/contracts/cross-chain/wormhole/L1BTCDepositorNttWithExecutor.sol

```diff
@@ -0,0 +1,648 @@
+        emit UnderlyingNttManagerUpdated(oldManager, _newNttManager);
+    }
+
+    /// @notice Updates the NTT Manager With Executor address
+    /// @param _newNttManagerWithExecutor New NTT Manager With Executor address
+    function updateNttManagerWithExecutor(address _newNttManagerWithExecutor)
+        external
+        onlyOwner
+    {
+        require(
+            _newNttManagerWithExecutor != address(0),
+            "Address cannot be zero"
+        );
+        address oldManager = address(nttManagerWithExecutor);
+        nttManagerWithExecutor = INttManagerWithExecutor(
+            _newNttManagerWithExecutor
+        );
+        emit NttManagerWithExecutorUpdated(
+            oldManager,
+            _newNttManagerWithExecutor
+        );
+    }
+
+    /// @notice Allows the owner to retrieve tokens from the contract
+    /// @param _token The address of the token to retrieve (address(0) for native token)
+    /// @param _to The address to send the tokens to
+    /// @param _amount The amount of tokens to retrieve
+    function retrieveTokens(
+        address _token,
+        address _to,
+        uint256 _amount
+    ) external onlyOwner {
+        require(
+            _to != address(0),
+            "Cannot retrieve tokens to the zero address"
+        );
+
+        if (_token == address(0)) {
+            payable(_to).transfer(_amount);
+        } else {
+            IERC20Upgradeable(_token).safeTransfer(_to, _amount);
+        }
+    }
+
+    /// @notice Quotes the cost using stored executor parameters
+    /// @dev Must call setExecutorParameters() first with real signed quote
+    /// @return cost Total cost for the transfer using stored parameters
+    function quoteFinalizeDeposit() external view returns (uint256 cost) {
+        require(
+            executorParametersSet,
+            "Must call setExecutorParameters() first with real signed quote"
+        );
+
+        // Extract destination chain from stored executor parameters
+        // We'll use the default chain since we don't have the receiver encoded in stored params
+        uint16 defaultChain = _getDefaultSupportedChain();
+        require(defaultChain != 0, "No supported chains configured");
+
+        return
+            nttManagerWithExecutor.quoteDeliveryPrice(
+                underlyingNttManager,
+                defaultChain,
+                "", // Empty transceiver instructions for basic transfer
+                storedExecutorArgs,
+                storedFeeArgs
+            );
+    }
+
+    /// @notice Sets executor parameters for the next finalizeDeposit call
+    /// @param executorArgs Real executor arguments with valid signed quote from Wormhole Executor API
+    /// @param feeArgs Fee arguments for the executor service
+    /// @dev Must be called before finalizeDeposit() to provide real signed quote
+    function setExecutorParameters(
+        ExecutorArgs memory executorArgs,
+        FeeArgs memory feeArgs
```

Comment:
Should we validate those?

Reply:
the quotes are NOT officially signed, I will change that name, they are economical estimations by the API.

Reply:
I will add an extra early check of the `msg.value` for the Executor Fees, otherwise this is checked in `NttManagerWithExecutor` and the error message is more confusing.

Reply:
```solidity
  require(
        msg.value >= executorArgs.value,
        "Insufficient payment for executor service"
    );
```

then also if hte NTT Executor say the Quote is Stale too low and reverts, the user can refresh the params and try again.